src/drivers/boards/common/board_crashdump.c

```diff
@@ -0,0 +1,172 @@
+#include <px4_config.h>
+#include <px4_tasks.h>
+
+#include <stdio.h>
+#include <string.h>
+
+#include <nuttx/board.h>
+
+#include "board_config.h"
+
+#include <systemlib/hardfault_log.h>
+
+#if defined(CONFIG_STM32_SAVE_CRASHDUMP)
+static void copy_reverse(stack_word_t *dest, stack_word_t *src, int size)
+{
+	while (size--) {
+		*dest++ = *src--;
+	}
+}
+#endif
+
+__EXPORT void board_crashdump(uintptr_t currentsp, FAR void *tcb, FAR const uint8_t *filename, int lineno)
+{
+	if (!hardfault_check_status) {
+		return;
+	}
+
+#if defined(CONFIG_STM32_SAVE_CRASHDUMP)
```

Comment:
> This will break the F7 builds. I think this Nuttx called function can be arch agnostic.

Ok, I will check this way to not break F7 builds: #if defined(CONFIG_STM32_SAVE_CRASHDUMP) || defined(CONFIG_STM32F7_SAVE_CRASHDUMP)

> Also if is not included nor called it should be optimized out. Did you find that was not the case?

Without even the function signature it will break the build.
```
#if defined(CONFIG_STM32_SAVE_CRASHDUMP)
int stm32_bbsram_savepanic(int fileno, uint8_t *context, int length);
#endif
```